api/controller/websiteControler.js

```diff
@@ -0,0 +1,36 @@
+const validator = require('../validate.js');
+var Message = require('../models/websiteModel.js');
+var myexpress = require("express")
+var router = myexpress.Router()
+module.exports = {
+
+    createWebsite:async (req, res) => {
+        try {
+          const validationRule = {
+            "cpu": [
+              "required",
+              "wedCpuTypes"
+            ],
+            "title": "required|string|min:3|max:50|EnglishLetters",
+            "description": "required|string|min:10|max:100|desEnglishLetters",
+            "domain":"isDomainAvailable",
+            "typeOfDomain":"tapedomin"
+          };
+          await validator(req.body, validationRule, {}, (err, status) => {
+            if (!status) {
+              res.status(412).send({
+                success: false,
+                message: 'Validation failed',
+                data: err
+              });
+            } else {
+              const message = new Message(req.body);
```

Comment:
don't create the website entity directly fron here.
use a service for it. (the validation process also need to be in the service).

Reply:
שינתי